### PR TITLE
MKL functions are enabled by default

### PR DESCRIPTION
**Pull request by prutskov**
Macro **ONEAPI_DAAL_USE_MKL_GPU_FUNC** was replaced by **ONEAPI_DAAL_NO_MKL_GPU_FUNC** with inversing of benaviour. MKL functions are used by default now.

It is fix isuue #245

### CODE REVIEW

**Comment by prutskov** [comment]
PCm: https://ecmd.inn.intel.com/commander/link/jobDetails/jobs/ea338875-025f-f172-8bb3-8cdcd4b723a1?s=Jobs?firstPage=1

**Comment by prutskov** [comment]
PCm: http://intel-ci.intel.com/ea35cfcf-7d50-f1d1-aee4-8cdcd4b723a1?

**Comment by prutskov** [comment]
restart

File: include/oneapi/internal/math/blas_executor.h [hunk 1]
```diff
@@ -24,13 +24,7 @@
 //--
 */
 
-#ifdef ONEAPI_DAAL_USE_MKL_GPU_FUNC
```
Comment (rlnx) [inline include/oneapi/internal/math/blas_executor.h]: Why did you remove this condition? MKL is expected to work only with clang. I would replace it with check for Intel DPC++ compiler not only for clang

File: include/oneapi/internal/math/blas_executor.h [hunk 2]
```diff
@@ -24,13 +24,7 @@
...
-    #if !(defined(__clang__))
-        #undef ONEAPI_DAAL_USE_MKL_GPU_FUNC
-    #endif
-#endif
```
Comment (rlnx) [inline include/oneapi/internal/math/blas_executor.h]: Why did you remove this condition? MKL is expected to work only with clang. I would replace it with check for Intel DPC++ compiler not only for clang

Comment (rlnx) [reply include/oneapi/internal/math/blas_executor.h]: Have a look at [this line](https://github.com/intel/daal/blob/master/include/services/daal_defines.h#L102), it checks for Intel DPC++ compiler `__SYCL_COMPILER_VERSION`

**Comment by prutskov** [comment]
last PCm: PCm: http://intel-ci.intel.com/ea36b6cd-1a6c-f168-890c-8cdcd4b723a1?

File: include/oneapi/internal/math/blas_executor.h (hunk 2 above)
Comment (prutskov) [reply include/oneapi/internal/math/blas_executor.h]: fixed
